aws/autoscaler/main.tf

```diff
@@ -1,40 +1,36 @@
-data "aws_eks_cluster" "cluster" {
-  name = var.cluster_name
-}
-
 data "aws_eks_node_group" "node_group" {
   cluster_name    = var.cluster_name
-  node_group_name = "${var.cluster_name}-deployment-node-group"
+  node_group_name = var.managed_node_group_name
 }
 
 locals {
-  cluster_oidc_issuer_url = data.aws_eks_cluster.cluster.identity[0].oidc[0].issuer
   autoscaling_group_name = data.aws_eks_node_group.node_group.resources.0.autoscaling_groups.0.name
 }
 
-module "oidc-provider-data" {
-  source = "../oidc-provider-data"
-  cluster_name = var.cluster_name
+# https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/aws#common-notes-and-gotchas
+resource "null_resource" "autoscaling_settings" {
+  provisioner "local-exec" {
+    interpreter = ["/bin/bash", "-c"]
+    command     = <<-SCRIPT
+      aws autoscaling suspend-processes --auto-scaling-group-name ${local.autoscaling_group_name} --scaling-processes AZRebalance
+      aws autoscaling update-auto-scaling-group --auto-scaling-group-name ${local.autoscaling_group_name} --default-cooldown 60
+      aws autoscaling enable-metrics-collection --auto-scaling-group-name ${local.autoscaling_group_name} --granularity "1Minute"
+    SCRIPT
+  }
 }
 
 module "eks-cluster-autoscaler" {
   source  = "lablabs/eks-cluster-autoscaler/aws"
-  version = "1.6.2"
+  version = "2.2.0"
 
   cluster_name                     = var.cluster_name
-  cluster_identity_oidc_issuer     = local.cluster_oidc_issuer_url
-  cluster_identity_oidc_issuer_arn = module.oidc-provider-data.arn
+  cluster_identity_oidc_issuer     = var.cluster_oidc_issuer_url
+  cluster_identity_oidc_issuer_arn = var.oidc_provider_arn
 
-
-  helm_chart_version = "9.21.0"
+  # make sure that chart version matches the cluster version
+  helm_chart_version = "9.33.0"
 
   values = yamlencode({
-    # Determined by our cluster version -
-    # Open the Cluster Autoscaler releases page from GitHub in a web browser and find the latest Cluster Autoscaler version that matches the Kubernetes major and minor version of your cluster.
-    # For example, if the Kubernetes version of your cluster is 1.20, find the latest Cluster Autoscaler release that begins with 1.20.
-    "image" : {
-      "tag" : "v${data.aws_eks_cluster.cluster.version}.0"
-    }
```

Comment:
the image should be in sync with the chart version, so it's better not to override it here.